Observer/Customer.php

```diff
@@ -43,8 +43,20 @@ private function getCustomerFromEvent($observer) {
                 break;
             case 'newsletter_subscriber_save_after':
                 $subscriber = $observer->getEvent()->getSubscriber();
-                if ($subscriber->isStatusChanged()) {
-                    return $this->metriloCustomer($this->customerRepository->getById($subscriber->getCustomerId()));
+                $customerId = $subscriber->getCustomerId();
+                if ($subscriber->isStatusChanged() && $customerId !== 0) {
+                    return $this->metriloCustomer($this->customerRepository->getById($customerId));
+                } else {
+                    $subscriberEmail = $subscriber->getEmail();
+                    return new MetriloCustomer(
+                        $subscriber->getStoreId(),
+                        $subscriberEmail,
+                        strtotime($subscriber->getData('change_status_at')) * 1000,
+                        $subscriberEmail,
+                        $subscriberEmail,
+                        true,
+                        ['guestCustomer']
```

Comment:
if this is the name of a tag, are you sure we use camel case for those?

Reply:
Tested this on production and there is no problem with case formatting, but API doc's are using snake case. Should I change it to 'guest_customer' ?

Reply:
yup.

Reply:
Done.